### PR TITLE
Adding configuration which updates traffic class on the basis of how …

### PR DESCRIPTION
**Pull request by ankitdixit**
…far the stream has caught up

### CODE REVIEW

File: logdevice/server/ServerSettings.cpp [hunk 1]
```diff
@@ -590,6 +590,29 @@ void ServerSettings::defineSettings(SettingEasyInit& init) {
...
+       if (val.count() <= 0) {
+         throw boost::program_options::error(
+           "traffic-class-updation-threshhold must be positive"
+         );
+       }
+     },
+     "If enabled, traffic classes of read stream are update automatically "
+     "on the basis of how much they have caught up",
+     SERVER,
+     SettingsCategory::Configuration)
```
Comment (mcrnic) [inline logdevice/server/ServerSettings.cpp:615]: Can we make it one setting automatic-traffic-class-selection-threshold and make 0 as disabled?

File: logdevice/server/read_path/ServerReadStream.h [hunk 2]
```diff
@@ -255,6 +257,16 @@ class ServerReadStream : boost::noncopyable {
...
+  void updateTrafficClass(
+      std::chrono::milliseconds traffic_class_updation_threshhold) {
+    if (last_delivered_timestamp_ + traffic_class_updation_threshhold >
+        SystemTimestamp::now().toMilliseconds()) {
+      traffic_class_ = TrafficClass::READ_TAIL;
+    } else {
+      traffic_class_ = TrafficClass::READ_BACKLOG;
+    }
+  }
+
```
Comment (mcrnic) [inline logdevice/server/read_path/ServerReadStream.h:269]: changes should only be applied if traffic_class_ was not previously set by client.
We don't want automation to override client set values

**Review by mcrnic — CHANGES_REQUESTED** [review]

File: logdevice/server/read_path/ServerReadStream.h (hunk 2 above)
Comment (ankitdixit) [reply logdevice/server/read_path/ServerReadStream.h:269]: Can you please help me with how to do this?
should I add a variable which remembers if "setTrafficClass()" has  been called?

File: logdevice/server/ServerSettings.cpp (hunk 1 above)
Comment (ankitdixit) [reply logdevice/server/ServerSettings.cpp:615]: done

**Pull request closed without merging**